src/app/IntervalTimer/interval-timer.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, Input, OnDestroy, OnInit, Output, EventEmitter } from '@angular/core';
+
+
+@Component({
+  selector: 'interval-timer',
+  templateUrl: './interval-timer.component.html',
+  styleUrls: ['./interval-timer.component.scss']
+})
+export class IntervalTimer implements OnInit, OnDestroy {
+
+  // Output to emit an Event everytime the Timer reaches 0
+  @Output()
+  intervalTimerEvent: EventEmitter<any> = new EventEmitter()
+
+  // Optional parameter to define the interval times that are needed in the parent Component. This overwrites the default of 10, 30, 60, 120 and 300 seconds if provided
+  @Input()
+  delayOptionsArray: number[] = [10, 30, 60, 120, 300];
+
+  // Optional parameter to provide another label for the Timer Component than "Refresh Rate"
+  @Input()
+  label: string = "Refresh Rate";
+
+  public callInterval: any;
+  public currentCallDelay: number;
+  public circleVisible: boolean = true;
+  public delayOptions: Generator; 
+
+
+  ngOnInit(): void {    
+    function* cycle<T>(array: T[]){
+      while (true) yield* array;
+    } 
+    this.delayOptions = cycle(this.delayOptionsArray)    
+    this.currentCallDelay = this.delayOptions.next().value;
+    // Establish timeout interval, get the named reference for it to update it later
+    this.callInterval = setInterval(() => {
+      this.intervalTimerEvent.emit(null);
+     } , this.currentCallDelay * 1000);
+  }
+
+
+  changeDelay() {
+    //Set next value of the interval array as call delay
+    this.currentCallDelay = this.delayOptions.next().value;
+  
+    clearInterval(this.callInterval);
+
+    this.callInterval = setInterval(() => {
+      this.intervalTimerEvent.emit(null);
+     } , this.currentCallDelay * 1000);
+
+     //Reload the Circle to refresh the Animation
+    this.circleVisible = false;
+    setTimeout(() => {      
+      this.circleVisible = true;
+    },0);
+    // Emit the timer Event on click
+    this.intervalTimerEvent.emit(null);
```

Comment:
If you separate this from the rest of the `changeDelay` method you can re-use it in `ngOnInit`. You could then instead bind a new method to the click event:

```ts
onTimerClick() {
  this.changeDelay()
  this.intervalTimerEvent.emit();
}
```